plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/animators/PuckPulsingAnimator.kt

```diff
@@ -19,19 +22,35 @@ internal class PuckPulsingAnimator : PuckAnimator<Double>(Evaluators.DOUBLE) {
   }
 
   fun animateInfinite() {
-    animate(0.0, maxRadius)
+    if (maxRadius <= 0.0) {
+      maxRadius = DEFAULT_RADIUS * pixelRatio
+    }
+    if (!isRunning) {
+      animate(0.0, maxRadius)
+    }
+    addListener(object : AnimatorListenerAdapter() {
+      override fun onAnimationRepeat(animation: Animator?) {
+        super.onAnimationRepeat(animation)
+        setObjectValues(0.0, maxRadius)
+      }
+    })
   }
 
   override fun updateLayer(fraction: Float, value: Double) {
     var opacity = 1.0f
     if (pulseFadeEnabled) {
-      opacity = 1.0f - (value / maxRadius).toFloat()
+      opacity = (1.0f - (value / maxRadius).toFloat()).coerceIn(0.0f, 1.0f)
```

Comment:
is coerceIn actually needed to cover the case with `maxRadius =-1`?

Reply:
Yes.